feat(region): adds endpoints for Region page support

ClemRz opened the pull request:
# feat(region): adds endpoints for Region page support

## 🤔 What

- Added complete Region page functionality with API endpoints integration
- Implemented Region-specific actions and reducers for data management
- Added subscription support for regions (subscribe/unsubscribe functionality)
- Created RegionsList component for displaying regions within countries
- Added data quality tracking for region entrances
- Integrated region statistics dashboard
- Added translation keys for region-related UI elements
- Cleaned up IDE configuration files (.idea folder removal)
- Added project documentation and translation management scripts
- Fixed massif creation / modification
- Fixed entrance permanent deletion

## 🤷♂️ Why

This change enables users to browse and interact with geographical regions within countries on GrottoCenter, providing a more granular way to explore cave data by administrative divisions. The region page allows users to view statistics, data quality metrics, and subscribe to updates for specific regions.

## 🔍 How

- **API Integration**: Added new action creators for region endpoints (`GetRegion`, `GetStatisticsRegion`, `GetEntrancesDataQuality`)
- **State Management**: Created dedicated reducers for region data, statistics, and subscription management
- **UI Components**: Built Region page component and RegionsList for country pages
- **Routing**: Added `/region/:id` route in App.jsx
- **Subscriptions**: Extended subscription system to support region-level notifications
- **Data Quality**: Integrated entrance data quality metrics specific to regions
- **Translations**: Added comprehensive translation support across en/es/fr languages

## 🔗 Related API PR

[feat(region): adds endpoints for Region page support #1395](https://github.com/GrottoCenter/grottocenter-api/pull/1395)

## 🧪 Testing

- Navigate to a country page and verify regions are listed
- Click on a region to access the new region page
- Test subscription/unsubscription functionality for regions
- Verify statistics and data quality metrics display correctly
- Test translation keys in different languages (en/es/fr)
- Go to My Account and test unsubscription functionality for regions
- Test massif creation modification
- Test entrance permanent deletion

## 📸 Previews

<img width="1511" height="886" alt="Screenshot 2025-11-11 at 8 19 02 a m" src="https://github.com/user-attachments/assets/06919913-a487-417a-b39d-cadf01151f9f" />
<img width="1097" height="733" alt="Screenshot 2025-11-11 at 8 18 52 a m" src="https://github.com/user-attachments/assets/f1e65dfa-b193-42bd-9e82-d2d2a8e14cbc" />
<img width="1347" height="532" alt="Screenshot 2025-11-11 at 8 18 29 a m" src="https://github.com/user-attachments/assets/2da5c073-7892-4417-bc66-dbc1cc6f6a75" />

Commit:
feat(region): adds Region page support

urien commented:
**Navigate to a country page and verify regions are listed** OK, pour la France on a les départements et les régions, c'est parfait

**Click on a region to access the new region page**: cela fonctionne. **Sur la page avec les informations détaillées peux tu mettre le nom de la région.Actuellement il y a le nom du pays et le mot région**

Commit:
fix(map): fixes blank map on sensitive entrances, removes the unsubscription by admins on another person's behalf (backend is not ready for that).

urien commented:
**Test subscription/unsubscription functionality for regions** cela fonctionne

**Verify statistics and data quality metrics display correctly** c'est bon, sur un échantillon... assez réduit

urien commented:
**Test translation keys in different languages (en/es/fr)** c'est ok

urien commented:
**Go to My Account and test unsubscription functionality for regions** c'est ok

urien commented:
**Test entrance permanent deletion** c'est tout bon

urien commented:
**Edit massif** OK

urien commented:
**Création d'un massif**: ok https://orange-rock-0d4f87503-1084.westeurope.azurestaticapps.net/ui/massifs/625

urien commented:
Si tu as le temps il faudrait enlever cette région https://orange-rock-0d4f87503-1084.westeurope.azurestaticapps.net/ui/countries/ES/regions/A
Elle n'existe pas

urien commented:
Le nom ne charge pas
<img width="1635" height="253" alt="{66EC3204-0BDE-41C5-89F0-C4969A223C90}" src="https://github.com/user-attachments/assets/f9b2e779-93ee-4eba-a0d9-58ae60e1b1f6" />

urien reviewed (approved):
Super, tout fonctionne

vmarseguerra commented on packages/web-app/package.json:
```diff
@@ -20,6 +20,7 @@
     "@mui/styles": "^5.16.7",
     "@mui/x-date-pickers": "^7.19.0",
     "@react-leaflet/core": "2.1.0",
+    "@tanstack/react-query": "^5.90.7",
```
Adding this dependency is unnecessary

ClemRz replied:
It is used in `packages/web-app/src/pages/ApplicationShell.jsx`. I thought it would make the app more robust. Let me know if you think it's not worth it.

ClemRz replied:
Actually let me remove it. It is not matching the general pattern.

vmarseguerra reviewed (changes requested):
Checking for translation introduces a lot of changes to many files. For a clean history, it is better to make them in a new PR or a separate commit

Commit:
fix(massif): fixes a bug where massifs cannot be created

Commit:
chore(translations): normalizes translations throughout the application

ClemRz commented:
@vmarseguerra I've reorganized the commits as you requested.

Commit:
fix(redux): remove undefined document reducer from store configuration

Commit:
fix: improve CSS selectors and form input handling

- Replace :nth-child with :nth-of-type for better specificity
- Add nullish coalescing for form field values to prevent controlled/uncontrolled input warnings
- Update PropTypes to match actual data structures
- Remove unused Input import from EntrancesSelector
- Fix duplicate key issue in RecentChangesCard
